File: ruby-advanced/lib/utils/validator.rb
module Utils
  class Validator
    EMAIL_REGEX = /\A[^\s@]+@[^\s@]+\.[^\s@]+\z/

    def self.valid_email?(email)
      !!(email =~ EMAIL_REGEX)
    end

    def self.valid_name?(name)
      name && name.length >= 2 && name.length <= 100
    end
  end
end

File: dotnet-advanced/Utils/Validator.cs
using System.Text.RegularExpressions;

namespace AdvancedUserService.Utils
{
    public class Validator
    {
        private static readonly Regex EmailRegex = new Regex(
            @"^[^\s@]+@[^\s@]+\.[^\s@]+$",
            RegexOptions.Compiled
        );

        public bool IsValidEmail(string email)
        {
            return !string.IsNullOrEmpty(email) && EmailRegex.IsMatch(email);
        }

        public bool IsValidName(string name)
        {
            return !string.IsNullOrEmpty(name) && name.Length >= 2 && name.Length <= 100;
        }
    }
}
